cube/src/main/java/zone/cogni/asquare/cube/index/ElasticsearchIndexingService.java

```diff
@@ -0,0 +1,604 @@
+                     .map(path -> StringUtils.substringBefore(path, "/")) // <- take folder name
+                     .distinct()
+                     .sorted()
+                     .map(collectionName -> calculateCollectionMetadata(indexMetadata, collectionName))
+                     .collect(Collectors.toList());
+  }
+
+  private CollectionMetadata calculateCollectionMetadata(IndexMetadata indexMetadata, String collection) {
+    String index = indexMetadata.getName();
+
+    CollectionMetadata result = new CollectionMetadata();
+    result.setName(collection);
+    result.setSelectQueryResources(getCollectionSelectQueries(index, collection));
+    result.setConstructQueryResources(getCollectionConstructQueries(index, collection));
+    result.setFacetQueryResources(getFacetQueries(index, collection));
+    return result;
+  }
+
+  private List<Resource> getCollectionSelectQueries(String index, String collection) {
+    return Arrays.asList(getResources("/" + index + "/" + collection + "/select-*.*"));
+  }
+
+  private List<Resource> getCollectionConstructQueries(String index, String collection) {
+    return Arrays.asList(getResources("/" + index + "/" + collection + "/construct-*.*"));
+  }
+
+  @PostConstruct
+  public void validate() {
+    this.indexesMetadata = calculateIndexesMetadata();
+
+    log.info("'{}' index service", configurationClasspath);
+
+    List<String> indexNames = getIndexNames();
+    log.info("{} index count: {}", indent, indexNames.size());
+
+    if (indexNames.isEmpty()) {
+      log.error("{} no valid indexes configured", indent);
+    }
+
+    if (!indexNames.isEmpty())
+      log.info("{} valid indexes:   {}", indent, String.join(", ", indexNames));
+    if (!getInvalidIndexNames().isEmpty())
+      log.info("{} invalid indexes: {}", indent, String.join(", ", getInvalidIndexNames()));
+
+    indexesMetadata.forEach(this::validateIndex);
+  }
+
+  private void validateIndex(IndexMetadata indexMetadata) {
+    // valid or not
+    if (indexMetadata.isValid())
+      log.info("  '{}' index is valid", indexMetadata.getName());
+    else
+      log.error("  '{}' index NOT is valid", indexMetadata.getName());
+
+    // settings
+    if (!indexMetadata.isValidSettingsResource())
+      log.error("{}   elastic-settings.json is missing", indent);
+
+    // collections
+    List<String> collectionNames = getValidCollectionNames(indexMetadata);
+    log.info("{}   collection count: {}", indent, collectionNames.size());
+
+    if (collectionNames.isEmpty()) {
+      log.error("{}   no valid collections configured", indent);
+    }
+
+    if (!collectionNames.isEmpty())
+      log.info("{}   valid collections:   {}", indent, String.join(", ", collectionNames));
+    if (!getInvalidCollectionNames(indexMetadata).isEmpty())
+      log.info("{}   invalid collections: {}", indent, String.join(", ", getInvalidCollectionNames(indexMetadata)));
```

Comment:
error log?

Reply:
error is already logged before: this is just informational in the context of an index.